discovery_engine/test/integration/request_feed_test.dart

```diff
@@ -63,20 +64,28 @@ void main() {
     });
 
     test(
-        'if requestNextFeedBatch fails due to a news api request error, requestFeed'
-        ' should return an empty list', () async {
+        'if requestNextFeedBatch traces a news api request error, then the'
+        ' depleted stacks have not been updated and subsequent calls should'
+        ' fail with FeedFailureReason.noNewsForMarket', () async {
```

Comment:
If the a trace is logged is not observable from the test so it should not be mentioned.
Returning an empty list is not an error.
If we want to return something we should signal an error iff we are completely unable to update the stacks.

Reply:
i'll reformulate the trace part.

regarding the return value, i thought the whole purpose of the task was to make `update_stacks()` infallible, should this be reverted as well?

the test for the empty list was just a side effect of the thrown dart exception due to the rust error in `DiscoveryEngineFfi.getFeedDocuments()`, which hasn't been caught before. now the `FeedManager.nextFeedBatch()` can progress further due to the removed exception and properly checks for `feed.isEmpty` to return the corresponding event, this event is now checked in the test.

Reply:
the aim was to make it fail less, having it infallible is better. If `getFeedDocuments` returns an error is ok to return `NextBatchRequestFailed` but then it must contain the error we get from `getFeedDocument` otherwise we are loosing information.
If `getFeedDocument` returns an empty list that is not an error.
@Robert-Steiner @biernacki

Reply:
changed as discussed: `update_stacks()` is fallible again, but will only fail if all stacks fail, it succeeds if at least one stack succeeds.

@Robert-Steiner this also means that #185 is still necessary.

Reply:
resolved merge conflicts, ready for rereview